server/re/src/msi_genquery2.cpp

```diff
@@ -0,0 +1,304 @@
+#include "irods/msi_genquery2.hpp"
+
+#include "irods/irods_at_scope_exit.hpp"
+#include "irods/rs_genquery2.hpp"
+#include "irods/msParam.h"
+#include "irods/irods_re_structs.hpp"
+#include "irods/irods_logger.hpp"
+#include "irods/irods_state_table.h"
+#include "irods/rodsError.h"
+#include "irods/rodsErrorTable.h"
+#include "irods/msi_preconditions.hpp"
+#include "irods/irods_exception.hpp"
+
+#include <fmt/format.h>
+#include <boost/any.hpp>
+#include <nlohmann/json.hpp>
+
+#include <algorithm>
+#include <cstdint>
+#include <cstdlib>
+#include <cstring>
+#include <iterator>
+#include <map>
+#include <string>
+
+namespace
+{
+    using log_msi = irods::experimental::log::microservice;
+
+    struct genquery2_context
+    {
+        nlohmann::json rows;
+        std::int32_t current_row = -1;
+    }; // struct genquery_context
+
+    // Holds the mappings for each GenQuery2 resultset.
+    std::map<std::size_t, genquery2_context> gq2_context; // NOLINT(cppcoreguidelines-avoid-non-const-global-variables)
+} // anonymous namespace
+
+/// Queries the catalog using the GenQuery2 parser.
```

Comment:
- Queries -> Query
- Add a line about contexts only being available to the agent that created it

Reply:
Done.